File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
         int left=0;
        int ans=0;
        map<char,int>mpp;
        for(int right=0;right<s.length();right++){
            mpp[s[right]]++;
            while(mpp[s[right]]>2){
                mpp[s[left]]--;
                left++;
            }
            ans=max(ans,right-left+1);
        }
        return ans;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
   Map<Character,Integer> mapp= new HashMap<>();
   int left=0;
   int ans =0;

for( int right=0;right<s.length();right++){
mapp.put(s.charAt(right),mapp.getOrDefault(s.charAt(right),0)+1);
while(mapp.get(s.charAt(right))>2){
    mapp.put(s.charAt(left),mapp.get(s.charAt(left))-1);
    left++;
}

ans= Math.max( ans,right-left+1);

}
return ans;



    }
}